osu.Game/OsuGameBase.cs

```diff
@@ -200,7 +200,7 @@ public void APIStateChanged(APIAccess api, APIState state)
             switch (state)
             {
                 case APIState.Online:
-                    LocalConfig.Set(OsuSetting.Username, LocalConfig.Get<bool>(OsuSetting.SaveUsername) ? API.Username : string.Empty);
+                    LocalConfig.Set(OsuSetting.Username, LocalConfig.Get<bool>(OsuSetting.SaveUsername) ? API.LocalUser.Value.Username : string.Empty);

```

Comment:
Rather than changing this here, I believe the value of `Username` should be updated earlier on (probably at line 129).